python/pyspark/sql/readwriter.py

```diff
@@ -1936,7 +1936,23 @@ def parquet(
         if partitionBy is not None:
             self.partitionBy(partitionBy)
         self._set_opts(compression=compression)
-        self._jwrite.parquet(path)
+        if self._df.isEmpty() is True and partitionBy is not None:
+            self._write_empty_partition(path, partitionBy)
+        else:
+            self._jwrite.parquet(path)
+    
+    @overload
+    def _write_empty_partition(self, path: str, partitionBy: str) -> None:
+        ...
+    
+    @overload
+    def _write_empty_partition(self, path: str, partitionBy: List[str]) -> None:
+        ...
+
+    def _write_empty_partition(self, path: str, partitionBy: Union[str, List[str]]) -> None:
+            temp_row_data = {col: None for col in partitionBy} if isinstance(partitionBy, list) else {partitionBy: None}
+            temp_df = self._df.sparkSession.createDataFrame([Row(**temp_row_data)])
+            temp_df.write.mode("overwrite").partitionBy(partitionBy).parquet(path)
```

Comment:
Does this work with Scala API too?

Reply:
Thanks for your review. No, this change only applies for PySpark, Scala API has no change with it. Do you think we should support it for Scala API?

Reply:
The change has to be made for API parity.

Reply:
I will change this PR to draft and work on changes for Scala API

Reply:
Hi @HyukjinKwon, I'm under the impression that PySpark relies on the underlying Scala API for Parquet operations. If that's correct, would updating the Scala API alone be sufficient to introduce this behavior to both PySpark and Scala API?